controllers/mariadbaccount_controller.go

```
@@ -0,0 +1,322 @@
+
+	// if not found because the label is missing, don't requeue, will be
+	// reconciled again when label is fixed
+	if err != nil && k8s_errors.IsNotFound(err) && instance.ObjectMeta.Labels["mariaDBDatabaseName"] == "" {
+		r.Log.Info(fmt.Sprintf("MariaDBAccount '%s' does not have a 'mariaDBDatabaseName' label; returning", instance.Name))
+		return ctrl.Result{}, err
+	}
+
+	// not found, or found but not in completed status.  requeue for a create,
+	// exit operation for a delete
+	if (err != nil && k8s_errors.IsNotFound(err)) || (err == nil && !mariadbDatabase.Status.Completed) {
+		if !isDelete {
+			// for the create case, need to wait for the MariaDBDatabase to exists before we can continue;
+			// requeue
+			if err != nil {
+				r.Log.Info(fmt.Sprintf(
+					"MariaDBAccount '%s' didn't find MariaDBDatabase '%s'; requeueing",
+					instance.Name, instance.ObjectMeta.Labels["mariaDBDatabaseName"]))
+			} else {
+				r.Log.Info(fmt.Sprintf(
+					"MariaDBAccount '%s' MariaDBDatabase '%s' not yet complete; requeueing",
+					instance.Name, instance.ObjectMeta.Labels["mariaDBDatabaseName"]))
+			}
+			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+		} else {
+			// for the delete case, the database doesn't exist.  so
+			// that means we don't, either.   remove finalizer from
+			// our own instance and return
+			if err != nil {
+				r.Log.Info(fmt.Sprintf(
+					"MariaDBAccount '%s' Didn't find MariaDBDatabase '%s'; no account delete needed",
+					instance.Name, instance.ObjectMeta.Labels["mariaDBDatabaseName"]))
+
+			} else {
+				r.Log.Info(fmt.Sprintf(
+					"MariaDBAccount '%s' MariaDBDatabase '%s' not yet complete; no account delete needed",
+					instance.Name, instance.ObjectMeta.Labels["mariaDBDatabaseName"]))
+			}
+			controllerutil.RemoveFinalizer(instance, helper.GetFinalizer())
+
+			return ctrl.Result{}, nil
+		}
+	} else if err != nil {
+		// unhandled error; exit
+		return ctrl.Result{}, nil
```

For troubleshooting purposes please at least log the error but also probably better to return the error